components/brave_wallet/browser/asset_discovery_manager.cc

```diff
@@ -331,18 +452,31 @@ void AssetDiscoveryManager::DiscoverAssetsOnAllSupportedChainsRefresh(
       from_block = *next_asset_discovery_from_block;
     }
 
-    DiscoverAssets(chain_id, mojom::CoinType::ETH, account_addresses, false,
-                   from_block, to_block);
+    DiscoverEthAssets(chain_id, mojom::CoinType::ETH, eth_account_addresses,
+                      false, from_block, to_block);
   }
 }
 
 void AssetDiscoveryManager::AccountsAdded(
     mojom::CoinType coin,
     const std::vector<std::string>& addresses) {
-  if (coin != mojom::CoinType::ETH || addresses.size() == 0u) {
+  if (!(coin == mojom::CoinType::ETH || coin == mojom::CoinType::SOL) ||
+      addresses.size() == 0u) {
     return;
   }
-  DiscoverAssetsOnAllSupportedChainsAccountsAdded(addresses);
+  DiscoverAssetsOnAllSupportedChainsAccountsAdded(coin, addresses);
+}
+
+// static
+absl::optional<std::string> AssetDiscoveryManager::DecodeMintAddress(
+    const std::vector<uint8_t>& data) {
+  if (data.size() < 32) {
+    return absl::nullopt;
+  }
+
+  std::vector<uint8_t> pub_key_bytes(data.begin(), data.begin() + 32);
+  std::string pub_key = Base58Encode(pub_key_bytes);
+  return pub_key;
```

Comment:
Please return as optional<SolanaAddress>. 
Nit: avoid temporary vector here and use base::span